ksql-rest-app/src/main/java/io/confluent/ksql/rest/server/execution/RequestHandler.java

```diff
@@ -69,28 +70,39 @@ public RequestHandler(
     this.ksqlEngine = Objects.requireNonNull(ksqlEngine, "ksqlEngine");
     this.ksqlConfig = Objects.requireNonNull(ksqlConfig, "ksqlConfig");
     this.serviceContext = Objects.requireNonNull(serviceContext, "serviceContext");
-    this.distributor = Objects.requireNonNull(distributor, "distributor");
+    this.distributorFactory = Objects.requireNonNull(distributorFactory, "distributorFactory");
     this.commandQueueSync = Objects.requireNonNull(commandQueueSync, "commandQueueSync");
   }
 
+  /**
+   * Executes the validated statements.
+   *
+   * @param statements        the statements to execute
+   * @param propertyOverrides any custom overrides
+   * @param verifyOffset      the verifyOffset under which the statements were validated
+   *
+   * @return entities representing the execution result
+   */
   public KsqlEntityList execute(
       final List<ParsedStatement> statements,
-      final Map<String, Object> propertyOverrides
-  ) {
+      final Map<String, Object> propertyOverrides,
+      final long verifyOffset) {
```

Comment:
Feels wrong to be passing the offset down here.

It almost feels like the engine should _know_ its own offset/checksum.  Otherwise there is a large amount of scope for the engine and offset to get out of whack.  

THoughts?